pkg/io/binaryWriter.go

```diff
@@ -19,8 +19,8 @@ func NewBinWriterFromIO(iow io.Writer) *BinWriter {
 	return &BinWriter{w: iow}
 }
 
-// WriteLE writes into the underlying io.Writer from an object v in little-endian format.
-func (w *BinWriter) WriteLE(v interface{}) {
+// writeLE writes into the underlying io.Writer from an object v in little-endian format.
+func (w *BinWriter) writeLE(v interface{}) {
```

Comment:
Why, oh why? When I'm writing some integers (not just arrays of bytes) I want to be very specific about LE/BE and thus it makes sense to have it exported.

Reply:
But we have exported WriteBytes and it actually does that. Why do we need WriteBytes and WriteLE both then?

Reply:
Because it's not obvious what `WriteBytes` is going to do with integers. It actually should only accept `[]byte` IMO, the same way `WriteVarBytes` does. And it can also be simplified to not use `writeLE` but just do `Write` because it operates on `[]byte` (it's just cheaper than going through `binary`).

Reply:
So in the end for `[]byte` we're going to have `WriteBytes` (that is plain and simple `Write` with error handling wrapper) and `WriteVarBytes` that writes fancy length prefix and then the same buffer of bytes. For integers we should have both `WriteLE` and `WriteBE` exposed to be used from outside because that's what you use for integers in various situations. Of course they still can also be misused for `[]byte`s, but so be it.

It also invalidates some part of changes made here where `WriteLE` is used for various integers, that usages are completely fine, valid and should stay.

Reply:
fixed